### ISSUE TITLE
Too many parameters created for Individual 

### ISSUE THREAD

**Issue by rwmcintosh** [Open-Systems-Pharmacology/PK-Sim#2656]
There are too many parameters added to individual building blocks when converting to a core building block.

Individual building block should only include species specific parameters.

**Comment by rwmcintosh** [comment]
These parameters are all excluded for being not species specific

| Parameter name|
|--------|
|SA proportionality factor
|Specific clearance (endosome)
|Surface area scaling exponent
|Weight
|Weight of blood organs
|Weight of tissue organs
|pH (blood cells)
|pH (interstitial)
|pH (intracellular)
|pH (plasma)
|Scaling exponent for fluid recirculation flow rate
|Thickness (endothelium)
|Fraction endosomal (global)
|Protein ratio (interstial/plasma)
|Vf (lipid, blood cells)
|Vf (lipid, plasma)
|Vf (neutral lipid, blood cells)-WS
|Vf (neutral lipid, plasma)-RR
|Vf (protein,blood cells)
|Vf (protein,plasma)
|Vf (water,blood cells)
|Vf (water,interstitial)
|Vf (water,plasma)
|Surface/Volume ratio (blood cells)
|Vf (neutral lipid, plasma)-PT
|Vf (neutral phospholipid, blood cells)-WS
|Vf (neutral phospholipid, plasma)-RR
|Fraction of endosomal uptake from plasma (global)
|Ontogeny factor (albumin)
|Ontogeny factor (alpha1-acid glycoprotein)
|Plasma protein scale factor
|Vf (acidic phospholipids, blood cells)-WS
|Vf (neutral lipid, blood cells)-RR
|Vf (phospholipid, plasma)-PT
|Vf (neutral phospholipid, blood cells)-RR
|Vf (water,plasma)-PT
|Fraction recycled to plasma (global)
|Acidic phospholipids (blood cells) [mg/g] - RR
|Vf (intracellular water, blood cells)-RR
|Rate constant for endosomal uptake (global)
|Rate constant for recycling from endosomal space (global)
|Hematocrit
|MeanBW
|Percentile
|Mean
|Deviation
|Weight (tissue)
|Density (tissue)
|Allometric scale factor
|Fraction vascular
|Volume
|Initial concentration
|Peripheral blood flow fraction
|Fraction endosomal
|Blood flow rate
|Fluid recirculation flow rate
|Lymph flow rate
|Volume (endothelium)
|Fraction intracellular
|Fluid recirculation flow proportionality factor
|Lymph flow proportionality factor
|Organ volume mouse
|Fraction of blood for sampling
|Flow fraction via large pores
|Hydraulic conductivity
|Radius (large pores)
|Radius (small pores)
|Vf (lipid)
|Vf (neutral lipid)-PT
|Vf (phospholipid)-PT
|Vf (protein)
|Fraction interstitial
|Albumin ratio (tissue/plasma)
|Albumin ratio (tissue/plasma)-PT
|Lipoprotein ratio (tissue/plasma)
|Vf (water)
|Vf (water)-PT
|Vf (neutral lipid)-RR
|Vf (neutral lipid)-WS
|Vf (neutral phospholipid)-RR
|Vf (neutral phospholipid, plasma)-WS
|Vf (extracellular water)-RR
|Vf (protein)-WS
|Vf (intracellular water)-RR
|Vf (water)-WS
|Acidic phospholipids [mg/g] - RR
|Vf (acidic phospholipids)-WS
|Specific blood flow rate
|pH
|Fraction expressed basolateral
|Relative expression
|Fraction expressed at blood brain barrier
|Fraction expressed brain tissue
|GeometricDeviation
|GFR (specific)
|Renal aging scaling factor
|fGFRpremat
|Fraction expressed apical
|Effective surface area variability factor
|Geometric surface area
|Volume (gut wall)
|Fractional steady state fill level
|pH in fed state
|Thickness of gut wall
|Inverse rate of inflow of liquid into stomach
|pH in fasted state
|Distal radius
|Gastric emptying time
|GET_alpha (Weibull function) variability factor
|GET_beta (Weibull function) variability factor
|Length
|Proximal radius
|Effective surface area
|Intestinal transit rate
|Microvilli factor
|Effective surface area enhancement factor
|Small intestinal transit time factor
|Fluid recirculation flow rate (incl. mucosa)
|Lymph flow rate (incl. mucosa)
|Volume (mucosa)
|Mucosa blood flow fraction
|Small intestinal transit time factor intercept
|Small intestinal transit time factor slope
|Small intestinal transit time
|Fraction mucosa
|Fraction of regional blood flow rate
|Large intestinal transit time factor
|Large intestinal transit time factor intercept
|Large intestinal transit time factor slope
|Large intestinal transit time
|Is liver zonated
|Fraction of periportal zone
|EHC continuous fraction
|Microsomal protein mass/g tissue
|Number of cells/g tissue
|Kd (FcRn, endogenous IgG) in plasma/interstitial
|kass (FcRn, endogenous IgG)
|Kd (FcRn, endogenous IgG) in endosomal space
|Radius (endogeneous IgG)
|Start concentration of free endogenous IgG (plasma)
|Start concentration of free FcRn (endosome)
|Surface area (blood cells/plasma)
|Surface area (interstitial/intracellular)
|Surface area (plasma/interstitial)
|Reference concentration
|t1/2 (liver)
|t1/2 (intestine)
|Ontogeny factor GI
|Ontogeny factor
|Relative expression in blood cells
|Relative expression in vascular endothelium

**Comment by msevestre** [comment]
>Observer 'Whole Blood' with path 'Whole Blood' references an entity with path 'simulation|Organism|Hematocrit' that cannot be found

Looks for hematocrit and it should be there. It's in the list of not being species specific. Too many parameter are filtered out

**Comment by msevestre** [comment]
@Yuri05 The implementation of the common parameters is wrong, or at least not complete

- Parameter such as Hematoctrit are marked as common where they are not (values is specific specific)
- Mean parameter of species dependendent parameter such as Mean of Age of aging onset are marked as common for all species which they are not
- Formula parameter with different formula for species (such as renal aging scaling factor) have species specific formulas but as marked as common
![image](https://github.com/Open-Systems-Pharmacology/PK-Sim/assets/1041237/70fa63d0-7d70-448c-983e-38ab3e3490c8)

**Comment by Yuri05** [comment]
"_Common parameters_" are those which are **present** for every species, not necessarily with the same value or the same formula.
That was the idea.

**Comment by msevestre** [comment]
Then this concept is missing something because how do we decide where to export ? Right now, we duplicate a lot of stuff

The mean parameter however is still a bug I think

**Comment by Yuri05** [comment]
* Parameters **with BB-Type Individual/Population**
  * Common parameters should be created both in the spatial structure and in the individual-BB
    * In the spatial structure with NaN value
    * In the Individual-BB: with their value or formula
  * Non-Common parameters should be created only in the individual-BB with their value or formula (and added in the simulation on the fly)

Something like this, no?

**Comment by Yuri05** [comment]
Regarding the Mean/STD parameters: they are not part of the PK-Sim DB and are created on the fly.
They should always get the same "common" state as their "parent" parameter?

**Comment by msevestre** [comment]
Ok so this repository returns true where it should return false or vice versa for parameters that it does not know. Probably the logic for unknown parameter should be it's NOT common

**Comment by msevestre** [comment]
>Common parameters should be created both in the spatial structure and in the individual-BB
In the spatial structure with NaN value
In the Individual-BB: with their value or formula

I don't know. Do we want to create the formula for BW = Sum of all organs weight as sum formula in ALL individuals? I thought this would be defined in the SP structure. Then it's ok but the individual really is huge and the spatial structure has only NaNs lol

**Comment by Yuri05** [comment]
> Ok so this repository returns true where it should return false or vice versa for parameters that it does not know. Probably the logic for unknown parameter should be it's NOT common

hmm, need to check this...

> Do we want to create the formula for BW = Sum of all organs weight as sum formula in ALL individuals?

yes! 

>I thought this would be defined in the SP structure. 

yes! BW is common parameter, right?

**Comment by msevestre** [comment]
Then I don't understand what we are trying to do lol
What's the point of defining BW in the individual with a formula that is the same for all species? 

And why adding parameters with NAN in your structure?

**Comment by Yuri05** [comment]
> And why adding parameters with NAN in your structure?

That's something we have discussed some time ago... 
But I also don't see a problem, if we export individual parameters **with their current value/formula** to the spatial structure instead of NaN.
@PavelBal

**Comment by msevestre** [comment]
I understand when there is a parameter with different formulas. But with the same formulas, I am just highly confused about what we are trying to do

**Comment by PavelBal** [comment]
21.06.2023: Proposal to put all parameters that are defined by a formula AND the formula is identical for all species - put the parameter into the spatial structure.

**Comment by PavelBal** [comment]
Is this clear by now, or do we need additional discussion?

**Comment by rwmcintosh** [comment]
- Parameters that are present in all species with different value or formula => Spatial structure as NaN, defined in individual
- Parameters that are present in all species with same value or formula => Spatial structure
- Parameter not present in all species => Individual

**Comment by Yuri05** [comment]
I have created another issue for the DB query which is required for this one: https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2699
Once the PR for that issue is merged - this one can be fixed (not by me then 😉 )
